irohad/network/impl/peer_tls_certificates_provider_wsv.cpp

```diff
@@ -47,11 +47,11 @@ Result<TLSCertificateType, std::string> PeerTlsCertificatesProviderWsv::get(
 }
 
 Result<TLSCertificateType, std::string> PeerTlsCertificatesProviderWsv::get(
-    const shared_model::interface::types::PubkeyType &public_key) const {
+    shared_model::interface::types::PublicKeyHexStringView public_key) const {
   auto opt_peer = impl_->getPeerFromWsv(public_key);
   if (not opt_peer) {
     return makeError(std::string{"Could not find peer by "}
-                     + public_key.toString());
+                     + std::string{public_key});
```

Comment:
I've made our logger use 'iroha::to_string', which is now the single point for producing printable representations.
it would be helpful to add a macro or any simple way to let `iroha::to_sintrg` work with strong types. maybe something like
```c++
#define IROHA_STRONG_STRING_VIEW_TYPE_TO_STRING(name)                  \
  namespace iroha {                                                    \
    namespace to_string {                                              \
      inline std::string ::iroha::to_string::toString(const name &o) { \
        return ::fmt::format("[name: {}]", std::string_view{o});       \
      }                                                                \
    }                                                                  \
  }
```

Reply:
I think we can postpone this change in order to avoid macros. There might be a better alternative to boost strong typedef, but it works for now.

Reply:
my point is that we should contain the changes in some fixed location rather then scattered throughout the project. whatever alternative to boost strong types we find, we should enable it to work with logger seamlessly, so that the only change we have to make is in a fixed set of helper macros or overloads whatever. but provided your current approach, we will eventually add more logging of strong types with in-place "workarounds".

Reply:
OK, this idea was not clear from the previous comment.